addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/CrossSiteScriptingScanRule.java

```diff
@@ -588,16 +588,18 @@ public void scan(HttpMessage msg, String param, String value) {
                             if (contexts2 == null) {
                                 return;
                             }
-                            if (contexts2.size() > 0) {
-                                // Yep, its vulnerable
-                                newAlert()
-                                        .setConfidence(Alert.CONFIDENCE_MEDIUM)
-                                        .setParam(param)
-                                        .setAttack(contexts2.get(0).getTarget())
-                                        .setEvidence(contexts2.get(0).getTarget())
-                                        .setMessage(contexts2.get(0).getMsg())
-                                        .raise();
-                                attackWorked = true;
+                            for (HtmlContext ctx : contexts2) {
+                                if (contexts2.size() > 0 && ctx.getSurroundingQuote().isEmpty()) {
```

Comment:
- The size check is redundant here.
 - This will lead to more alerts being raised than before (assuming more than one context) and always with the details of the first context. Do you meant to add a break?

Reply:
Now I added size check before the for loop. I think this will not raise more alerts if there are more than one context.

Reply:
It will still raise them, it would have to break the loop to just raise once (technically the size check is no longer needed with the use of the loop).

Reply:
Also, worth trying to add a test for this case.

Reply:
Here context is the reflection of payload in HTML right?

Reply:
Yeah.

Reply:
With or without size check,  It will not give multiple alerts even If there are more than one context. However size check is not required so I will remove it.

Reply:
You mean with `contexts2` having size more than 2?

Reply:
Yes.

Reply:
And both having no surrounding quote?

Reply:
Yeah.

Reply:
So, the break was added, that makes more sense.